tests/test_modeling_common.py

```diff
@@ -3179,9 +3179,9 @@ def test_mismatched_shapes_have_properly_initialized_weights(self):
                 continue
 
             # TODO: ydshieh
-            if model_class.__name__ in ["Wav2Vec2ForSequenceClassification", "Wav2Vec2ForSequenceClassification"]:
+            if model_class.__name__ in ["Wav2Vec2ForSequenceClassification", "Wav2Vec2ForSequenceClassification", "CLIPForImageClassification", "RegNetForImageClassification"]:
```

Comment:
Hmmm - skipping a test because it's failing it's the best reason....  As this is already done, I'm OK with it providing there's an issue to track so this is resolved.

Reply:
@amyeroberts Here there are 2 kinds of failures:

- one is about not initialized at all and giving very large or even `nan` values: this is important and should be fixed
- one is about if we are using `config.initializer_factor` to control the initialization, but that is really more for testing purpose (historically)

```
        initializer_factor (`float`, *optional*, defaults to 1.0):
            A factor for initializing all weight matrices (should be kept to 1, used internally for initialization
            testing).
```
and that is less important. For the model classes I added here, they seem all belong to the 2nd cases.

Reply:
Regarding  #30451, it might contain other issues/cases I don't observed here though.